django_declarative_apis/machinery/filtering.py

```diff
@@ -67,8 +71,12 @@ def _get_filtered_field_value(
 
     if isinstance(field_type, types.FunctionType):
         val = field_type(inst)
-    elif isinstance(field_type, _ExpandableForeignKey) and not expand_this:
-        val = _get_unexpanded_field_value(inst, field_name, field_type)
+    elif isinstance(field_type, _ExpandableForeignKey):
+        if expand_this:
+            inst_field_name = field_type.inst_field_name or field_name
+            val = getattr(inst, inst_field_name)
+        else:
+            val = _get_unexpanded_field_value(inst, field_name, field_type)
```

Comment:
This doesn't need the same `inst_field_name` treatment because it's not actually hitting the model, right?

Reply:
it does need it, and accesses it within the called function as `field_type.inst_field_name`